Makefile

```diff
@@ -1,157 +1,282 @@
+	./hack/update-codegen.sh v${CODE_GENERATOR_VERSION}
 
 .PHONY: generate-helm-docs
-generate-helm-docs:
-	helm-docs -s file -c deploy/charts/ -t README.md.gotmpl
+generate-helm-docs: helm-docs ## Generate Helm chart documentation
+	$(HELM_DOCS) -s file -c deploy/charts/ -t README.md.gotmpl
 
-deps: bin/golangci-lint bin/licensei bin/kind bin/kurun bin/controller-gen bin/helm-docs
-deps: ## Install dependencies
+.PHONY: generate
+generate: generate-manifests generate-code generate-helm-docs
+generate: ## Generate manifests, code, and docs resources
 
-bin/golangci-lint:
-	@mkdir -p bin
-	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | bash -s -- v${GOLANGCI_VERSION}
-
-bin/licensei:
-	@mkdir -p bin
-	curl -sfL https://raw.githubusercontent.com/goph/licensei/master/install.sh | bash -s -- v${LICENSEI_VERSION}
-
-bin/kind:
-	@mkdir -p bin
-	curl -Lo bin/kind https://kind.sigs.k8s.io/dl/v${KIND_VERSION}/kind-$(shell uname -s | tr '[:upper:]' '[:lower:]')-$(shell uname -m | sed -e "s/aarch64/arm64/; s/x86_64/amd64/")
-	@chmod +x bin/kind
-
-bin/kurun:
-	@mkdir -p bin
-	curl -Lo bin/kurun https://github.com/banzaicloud/kurun/releases/download/${KURUN_VERSION}/kurun-$(shell uname -s | tr '[:upper:]' '[:lower:]')-$(shell uname -m | sed -e "s/aarch64/arm64/; s/x86_64/amd64/")
-	@chmod +x bin/kurun
-
-bin/controller-gen:
-	@mkdir -p bin
-	set -ex ;\
-	CONTROLLER_GEN_TMP_DIR=$$(mktemp -d) ;\
-	cd $$CONTROLLER_GEN_TMP_DIR ;\
-	go mod init tmp ;\
-	GOBIN=$(PWD)/bin/controller-gen-${CONTROLLER_GEN_VERSION}-bin/ go install sigs.k8s.io/controller-tools/cmd/controller-gen@v${CONTROLLER_GEN_VERSION} ;\
-	mv $(PWD)/bin/controller-gen-${CONTROLLER_GEN_VERSION}-bin/controller-gen $(PWD)/bin/controller-gen ;\
-	rm -rf $(PWD)/bin/controller-gen-${CONTROLLER_GEN_VERSION}-bin ;\
-	rm -rf $$CONTROLLER_GEN_TMP_DIR
-
-bin/helm-docs:
-	@mkdir -p bin
-	curl -L https://github.com/norwoodj/helm-docs/releases/download/v${HELM_DOCS_VERSION}/helm-docs_${HELM_DOCS_VERSION}_$(shell uname)_x86_64.tar.gz | tar -zOxf - helm-docs > ./bin/helm-docs
-	@chmod +x bin/helm-docs
+##@ Build
 
-.PHONY: help
-.DEFAULT_GOAL := help
-help:
-	@grep -h -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-10s\033[0m %s\n", $$1, $$2}'
+.PHONY: build
+build: ## Build manager binary
+	@mkdir -p build
+	go build -race -o build/manager ./cmd
+
+.PHONY: run
+run: generate deploy ## Run the controller from your host
+	OPERATOR_NAME=vault-dev go run cmd/main.go -verbose
+
+# If you wish built the manager image targeting other platforms you can use the --platform flag.
+# (i.e. docker build --platform linux/arm64 ). However, you must enable docker buildKit for it.
+# More info: https://docs.docker.com/develop/develop-images/build_enhancements/
+.PHONY: docker-build
+docker-build: ## Build docker image
+	docker build -t ${IMG} .
+
+.PHONY: docker-push
+docker-push: ## Push docker image
+	docker push ${IMG}
```

Comment:
I don't think this is useful: people shouldn't be pushing images from their local environments.